kore/src/Kore/Rewrite/Strategy.hs

```diff
@@ -93,40 +84,20 @@ import GHC.Generics qualified as GHC
 import Kore.Rewrite.Transition
 import Numeric.Natural
 import Prelude.Kore hiding (
-    all,
-    and,
-    any,
-    many,
-    or,
     seq,
     sequence,
-    some,
  )
 
 {- | An execution strategy.
 
     @Strategy prim@ represents a strategy for execution by applying rewrite
     axioms of type @prim@.
 -}
-
--- TODO (thomas.tuegel): This could be implemented as an algebra so that a
--- strategy is a free monad over the primitive rule type and the result of
--- execution is not a generic tree, but a cofree comonad with exactly the
--- branching structure described by the strategy algebra.
 data Strategy prim where
-    -- The recursive arguments of these constructors are /intentionally/ lazy to
```

Comment:
This is probably out of scope for this PR, but we should probably keep in mind that we'd like to get rid of this type in the near future. Maybe a subsequent PR could take care of that?

Reply:
Yes, this type (and maybe even the whole module `Strategy`) is on the list of things to remove. I would think we lift everything worth keeping into the new `GraphTraversal` module.
Currently the `Prove` module still uses `Strategy` for the single-step function, and the repl uses the computation graph directly. Removing this and consolidating the parts we need from `Strategy` into `GraphTraversal` would be follow-up work in a new ticket and PR.